chore(samples): update Autoclass Sample for v2.1 features

BenWhitehead opened the pull request:
Depends on https://github.com/googleapis/java-storage/pull/2258

lesv reviewed (approved):
No test?  Otherwise LGTM

cojenco commented on samples/snippets/src/main/java/com/example/storage/bucket/SetBucketAutoclass.java:
```diff
@@ -21,32 +21,54 @@
+
+    // Note:  terminal_storage_class field is optional and defaults to NEARLINE if not otherwise
+    // specified. Valid terminal_storage_class values are NEARLINE and ARCHIVE.
     boolean enabled = false;
```
```suggestion
    boolean enabled = true;
```

cojenco commented on samples/snippets/src/main/java/com/example/storage/bucket/SetBucketAutoclass.java:
```diff
@@ -21,32 +21,54 @@
-    // To enable autoclass, you must enable it at bucket creation time.
+    // The storage class that objects in an Autoclass bucket eventually transition to if not read
+    // for a certain length of time
+    // StorageClass storageClass = StorageClass.NEARLINE;
```
```suggestion
    // StorageClass storageClass = StorageClass.ARCHIVE;
```

cojenco replied:
Apologies for the canonical sample changes, let's update this sample to set `enabled = true`  with `storageClass = StorageClass.ARCHIVE`. This is more aligned with the CUJ for setting terminal storage class. Thank you!

cojenco reviewed (commented):
Thanks for working on this! Could we also update the storage_get_autoclass sample to demonstrate the newly added fields?

https://github.com/googleapis/java-storage/blob/main/samples/snippets/src/main/java/com/example/storage/bucket/GetBucketAutoclass.java

BenWhitehead commented:
Now that libraries-bom has been released with the feature I've rebased and updated the sample and the samples test.

@cojenco I applied the value changes in the sample you recommended.

cojenco commented:
> Thanks for working on this! Could we also update the storage_get_autoclass sample to demonstrate the newly added fields?
> 
> https://github.com/googleapis/java-storage/blob/main/samples/snippets/src/main/java/com/example/storage/bucket/GetBucketAutoclass.java

LGTM, could we showcase the new fields in the get_autoclass sample too? Thank you @BenWhitehead

cojenco reviewed (approved):
lgtm, thanks Ben 🙂
